rsearch/conductor.go

```diff
@@ -0,0 +1,43 @@
+package rsearch
+
+func manageResources(ns Event, terminators map[string]chan Done, config Config, out chan Event) {
+	uid := ns.Object.Metadata.Uid
+	if ns.Type == "ADDED" {
+		done := make(chan Done)
+		terminators[uid] = done
+		ns.Object.Produce(out, terminators[uid], config)
+	} else if ns.Type == "DELETED" {
+		close(terminators[uid])
+		delete(terminators, uid)
+	} else if ns.Type == "_CRASH" {
+		for uid, c := range terminators {
+			close(c)
+			delete(terminators, uid)
+		}
+	}
+}
+
+// Conductor manages a set of goroutines one per namespace.
+func Conductor(in <-chan Event, done <-chan Done, config Config) <-chan Event {
+	// Idea of this map is to keep termination channels organized
+	// so when DELETED event occurs on a namespace it would be possible
+	// to terminater related goroutine
+	var terminators map[string]chan Done
+	terminators = make(map[string]chan Done)
+
+	ns := Event{}
+	out := make(chan Event)
+
+	go func() {
+		for {
+			select {
+			case ns = <-in:
```

Comment:
Technically yes. I just like it when i read it. I understand it's unnecessary, do you think it's also wrong thing to do ?